tests/test_pytypes.cpp

```diff
@@ -319,6 +319,24 @@ TEST_SUBMODULE(pytypes, m) {
         return a[py::slice(0, -1, 2)];
     });
 
+    // See #2361
+    // These four tests should reflect the text in `object.rst`, the
+    // "Interaction with None" section.
+    m.def("test_str_with_default_arg_none" ,[](py::str value) {
+        return value;
+    }, py::arg("value") = py::none());
+    m.def("test_str_assign_none", []() {
+        py::str is_this_none = py::none();
```

Comment:
This is not an assignment? This is a implicit conversion, I believe.

Reply:
To make a connection: #2366 will cover this, too (I still need to work on it though).
I think it's best to repeat the test in your context to also have complete coverage of the dimension you're sampling, but you could reuse the more general `convert_to_pybind11_str` function.
Don't let that hold up this PR if you're faster than me. I'll keep an eye out and could replace `test_str_assign_none` with the more general function later.

Reply:
> This is not an assignment? This is a implicit conversion, I believe.

Erp, well, actually, I think just a constructor. Implicit conversion I think would only get triggered if it were really an assignment...
Will change.

> To make a connection: #2366 will cover this, too (I still need to work on it though).

Sweet! Yeah, I will probably keep this here, and possibly try to exercise it among different types.

Reply:
Nope, you're right. It is an implicit copy constructor.

Reply:
Done.

Reply:
> Nope, you're right. It is an implicit copy constructor.
> 
> I always forget the difference between `T value = y`, `T value(y)`, and `T value{y}`, and how copy elision works...

Yup, same here. It's reasonably confusing how to call it, but anyway, it's clear what it does! :-)